relayer/src/link/relay_path.rs

```diff
@@ -1427,21 +1444,23 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
             return Ok(());
         }
 
-        info!(
-            "[{}] scheduling op. data with {} msg(s) for {} (height {})",
-            self,
-            od.batch.len(),
-            od.target,
-            od.proofs_height.increment(), // increment for easier correlation with the client logs
-        );
-
         // Update clients ahead of scheduling the operational data, if the delays are non-zero.
         if !self.zero_delay() {
+            debug!("[{}] connection delay is non-zero: updating client", self);
             let target_height = od.proofs_height.increment();
             match od.target {
-                OperationalDataTarget::Source => self.update_client_src(target_height)?,
-                OperationalDataTarget::Destination => self.update_client_dst(target_height)?,
+                OperationalDataTarget::Source => {
+                    self.update_client_src(target_height, &od.tracking_id)?
+                }
+                OperationalDataTarget::Destination => {
+                    self.update_client_dst(target_height, &od.tracking_id)?
+                }
             };
+        } else {
+            debug!(
+                "[{}] connection delay is zero: client update message will be prepended later",
+                self
+            );
```

Comment:
All this formatting of `self` maybe needs a more systematic alternative with a helper making use of spans and/or KVs, but I don't have a well formed proposal at the moment.